### ISSUE TITLE
Expo client app (managed workflow) on any Android emulator would build JavaScript 100% and quits automatically

### ISSUE THREAD

**Issue by ptkpyitheim** [expo/expo#14388]
### Summary

Expo client app (managed workflow) is what I have. I ran `expo start` and fired up the Android emulator. Expo client app would load and build JavaScript 100%. Then the client app crashes and quits automatically. API 29 emulator gives me "Expo Go Keeps Stopping" error. I've tried deleting all the caches, re-installing node_modules, restarting everything, installed everything else to the latest versions, etc.

Why is that happening? I couldn't find a duplicate problem on the internet.

### Managed or bare workflow? If you have made manual changes inside of the `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### Package versions

"expo": "~42.0.0",

### Environment

expo start

### Reproducible demo or steps to reproduce from a blank project

Run expo start. Open up an emulator through Android studio. Run on Android device/emulator on localhost:19002.

**Comment by lukmccall** [comment]
Hi @ptkpyitheim, thanks for your report. I would like to help you. However, I need more information to do it. Could you get logs from your emulator and post them somewhere? To do it, you need to simply run `adb logcat` when your emulator is running.  Here https://developer.android.com/studio/command-line/logcat your can find more information.

**Comment by ptkpyitheim** [comment]
[log.txt](https://github.com/expo/expo/files/7154516/log.txt)


I attached the log above. It's a pretty long file. But it seems like this line might be causing the problem ` Error opening kernel wakelock stats for: wakeup34: Permission denied`

**Comment by ptkpyitheim** [comment]
Or do you think this might be an issue on Android? I could try to fix this warning but it was working fine on iOS.` Require cycles are allowed, but can result in uninitialized values. Consider refactoring to remove the need for a cycle.`

**Comment by lukmccall** [comment]
I've managed to find an exception that causes the crash:
```
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: Error while updating prop padding
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: java.lang.reflect.InvocationTargetException
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at java.lang.reflect.Method.invoke(Native Method)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateShadowNodeProp(ViewManagersPropertyCache.java:7)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackShadowNodeSetter.setProperty(ViewManagerPropertyUpdater.java:2)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:14)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.ReactShadowNodeImpl.updateProperties(ReactShadowNodeImpl.java:1)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.UIImplementation.createView(UIImplementation.java:11)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.UIManagerModule.createView(UIManagerModule.java:5)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at java.lang.reflect.Method.invoke(Native Method)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:18)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:2)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at android.os.Handler.handleCallback(Handler.java:938)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at android.os.Handler.dispatchMessage(Handler.java:99)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:1)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at android.os.Looper.loop(Looper.java:223)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:8)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at java.lang.Thread.run(Thread.java:923)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: Caused by: java.lang.NumberFormatException: For input string: "3% 5% 0% 5"
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2043)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at sun.misc.FloatingDecimal.parseFloat(FloatingDecimal.java:122)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at java.lang.Float.parseFloat(Float.java:451)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.LayoutShadowNode$MutableYogaValue.setFromDynamic(LayoutShadowNode.java:11)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	at abi42_0_0.com.facebook.react.uimanager.LayoutShadowNode.setPaddings(LayoutShadowNode.java:4)
09-13 09:12:18.711  9010  9158 E unknown:ViewManager: 	... 17 more
```
what leads to:
```
09-13 09:12:18.712  9010  9158 F libc    : /usr/local/google/buildbot/src/android/ndk-release-r20/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:73: abort_message: assertion "terminating with uncaught exception of type facebook::jni::JniException: abi42_0_0.com.facebook.react.bridge.JSApplicationIllegalArgumentException: Error while updating property 'padding' in shadow node of type: RCTText" failed
09-13 09:12:18.713  9010  9158 F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 9158 (mqt_native_modu), pid 9010 (st.exp.exponent)
09-13 09:12:18.852  9187  9187 I crash_dump32: obtaining output fd from tombstoned, type: kDebuggerdTombstone
09-13 09:12:18.854   295   295 I tombstoned: received crash request for pid 9158
09-13 09:12:18.856  9187  9187 I crash_dump32: performing dump of process 9010 (target tid = 9158)
09-13 09:12:18.867  9187  9187 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
09-13 09:12:18.867  9187  9187 F DEBUG   : Build fingerprint: 'google/sdk_gphone_x86_arm/generic_x86_arm:11/RSR1.201013.001/6903271:userdebug/dev-keys'
09-13 09:12:18.867  9187  9187 F DEBUG   : Revision: '0'
09-13 09:12:18.868  9187  9187 F DEBUG   : ABI: 'x86'
09-13 09:12:18.869  9187  9187 F DEBUG   : Timestamp: 2021-09-13 09:12:18-0400
09-13 09:12:18.870  9187  9187 F DEBUG   : pid: 9010, tid: 9158, name: mqt_native_modu  >>> host.exp.exponent <<<
09-13 09:12:18.870  9187  9187 F DEBUG   : uid: 10153
09-13 09:12:18.870  9187  9187 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
09-13 09:12:18.871  9187  9187 F DEBUG   : Abort message: '/usr/local/google/buildbot/src/android/ndk-release-r20/external/libcxx/../../external/libcxxabi/src/abort_message.cpp:73: abort_message: assertion "terminating with uncaught exception of type facebook::jni::JniException: abi42_0_0.com.facebook.react.bridge.JSApplicationIllegalArgumentException: Error while updating property 'padding' in shadow node of type: RCTText" failed'
```

Unfortunately, I think there is something in your js code that leads to that exception. If you can share your project with me, I could dig deeper. If not, could you try to find a place in your code which causes that? After that share only the problematic bit of your code.

**Comment by ptkpyitheim** [comment]
I might know what the issue is. I'll try it out and update you. Thank you so much for your help so far.

**Comment by ptkpyitheim** [comment]
I figured it out. I used [React-Native-Responsive-Fontsize](https://www.npmjs.com/package/react-native-responsive-fontsize) library. And I changed all the paddings in my project using RFValue and/or RFPercentage from the library. For some reasons, Android doesn't like literal percentage values like `padding: "4%"`

**Comment by ipakhomov** [comment]
**TLDR**: If the fix [above](https://github.com/expo/expo/issues/14388#issuecomment-922374719) doesn't help, check logs for  `No identifiers allowed directly after numeric literal` error and downgrade [Color](https://www.npmjs.com/package/color) package to `v3.2.1` for a while if you are using it. 

I was facing a similar issue, but for me, the problem popped up after upgrading [Color](https://github.com/Qix-/color) from v3.2.1 to v4.0.1. After researching `adb logcat` logs I found the following error: `No identifiers allowed directly after numeric literal`.
Here are a couple of issues about this ([first](https://github.com/Qix-/color/issues/222) and [second](https://github.com/Qix-/color/issues/223/0), and turned out that this is an Android-specific issue, that is described [here](https://github.com/expo/expo/issues/12042). So looks like the core issue is related to the metro builder and opened [here](https://github.com/facebook/metro/issues/645) at the moment.
